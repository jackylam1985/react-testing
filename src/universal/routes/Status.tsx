import * as React from 'react'
import { Route, RouteComponentProps } from 'react-router-dom'

interface Props {
  status?: number
  children?: React.ReactNode
}

const render = (props: Props) => {
  return ({ staticContext }: RouteComponentProps<any>) => {
    if (staticContext) {
      staticContext.status = props.status
    }

    return props.children
  }
}

export default ({ status, children }: Props) => (
  <Route render={render({status, children})} />
)
