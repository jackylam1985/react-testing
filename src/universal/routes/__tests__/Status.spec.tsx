// tslint:disable-next-line no-unused-variable
import * as React from 'react'
import { createRenderer } from 'react-test-renderer/shallow'

import Status from '../../routes/Status'

describe('../routes/Status', () => {
  it('renders', () => {
    const renderer = createRenderer()
    expect(renderer.render(
      <Status status={404}/>,
    )).toMatchSnapshot()
  })
})
