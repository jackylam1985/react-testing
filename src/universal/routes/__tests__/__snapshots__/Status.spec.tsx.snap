// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`../routes/Status renders 1`] = `
<Route
  render={[Function]}
/>
`;
